lambdas/navigator_config/index.py

```diff
@@ -63,14 +80,55 @@ def config(dest_bucket, dest_dir, es_url, api_url, s3_bucket,
             'accessKeyId': '',
             'secretAccessKey': ''
         },
+        'registryUrl': 'https://' + registry_host + '/',
         'signInRedirect': '/',
         'signOutRedirect': '/'
     }
 
-    s3_client.put_object(
-        ACL='public-read',
-        Body=json.dumps(catalog_config),
-        Bucket=dest_bucket,
-        Key=dest_dir + 'config.json',
-        ContentType='application/json'
+    return (federation, catalog_config)
+
+def config(dest_bucket, dest_dir, es_url, api_url, s3_bucket,
+           bucket_title, bucket_icon, bucket_description,
+           registry_host):
+
+    def get_existing(key_name):
+        """
+        Get existing config if it exists, otherwise return a new empty dict.
+        """
+        try:
+            response = s3_client.get_object(Bucket=dest_bucket, Key=key_name)
+            return json.load(response['Body'])
+        except ClientError as e:
+            if e.response['Error']['Code'] == "NoSuchKey":
+                return {}
+            raise e
+
+    region = boto3.session.Session().region_name
```

Comment:
is `boto3.session.Session().region_name` the region that the current aws profile/cli is config'ed to? I'm worried that this might override a region if used against a bucket in a different region than the default configed one, but am not sure how this is used.

When does this actually get run?

Reply:
that gets run by the lambda task, which will be in the same region as the stack, which is what we want.

Reply:
cool, thanks for clarifying